pkg/ottl/expression.go

```diff
@@ -549,6 +551,88 @@ func (g StandardIntLikeGetter[K]) Get(ctx context.Context, tCtx K) (*int64, erro
 	return &result, nil
 }
 
+// ByteSliceLikeGetter is a Getter that returns []byte by converting the underlying value to an []byte if necessary
+type ByteSliceLikeGetter[K any] interface {
+	// Get retrieves []byte value.
+	// The expectation is that the underlying value is converted to []byte if possible.
+	// If the value cannot be converted to []byte, nil and an error are returned.
+	// If the value is nil, nil is returned without an error.
+	Get(ctx context.Context, tCtx K) ([]byte, error)
+}
+
+type StandardByteSliceLikeGetter[K any] struct {
+	Getter func(ctx context.Context, tCtx K) (any, error)
+}
+
+func (g StandardByteSliceLikeGetter[K]) Get(ctx context.Context, tCtx K) ([]byte, error) {
+	val, err := g.Getter(ctx, tCtx)
+	if err != nil {
+		return nil, fmt.Errorf("error getting value in %T: %w", g, err)
+	}
+	if val == nil {
+		return nil, nil
+	}
+	var result []byte
+	switch v := val.(type) {
```

Comment:
I think we can use `binary.Write` to encode all types we support. The easiest way I can think of to do it would be to make an `any`-typed variable that we assign to in each branch of the switch statement, then just call `binary.Write` on it.

Reply:
adapted it for boolean values, for string values it's not possible according to the `binary.Write()` docs:

```
// Data must be a fixed-size value or a slice of fixed-size
// values, or a pointer to such data.`
```

Since `string` in Golang is represented as a non-fixed-size slice, we cannot use it

Reply:
Good callout. That's fine, I think we have enough coverage with the other types to make it worth it.